### PR TITLE
Add config to make the bot auto-introduce new members

### PR DESCRIPTION
**Pull request by crclark96**
I'm back!

A month or two ago people were talking about getting rid of the `$intro` prompt to slackbot by automating it with bishop and so I did it - I put the intro message in `config.json`, let me know if that's the right place for it

### CODE REVIEW

File: server/botserver.py [hunk 1]
```diff
@@ -98,13 +98,16 @@ def parse_slack_message(self, message_list):
...
+                    previous_msg = msg['previous_message']['text']
                     delete_keywords = log_deletions.split(",")
 
                     if any(keyword.strip() in previous_msg for keyword in delete_keywords):
                         user_name = self.slack_wrapper.get_member(msg['previous_message']['user'])
                         display_name = get_display_name(user_name)
                         self.slack_wrapper.post_message(msg['channel'], "*{}* deleted : `{}`".format(display_name, previous_msg))
+            # Greet new users
+            elif msg.get("type") == "im_created":
+                self.slack_wrapper.post_message(msg['user'], self.get_config_option("intro_message"))
```
Comment (Grazfather) [inline server/botserver.py]: Could we make this conditional on there being an intro message set? What happens if it's blank?

Comment (crclark96) [reply server/botserver.py]: I tested this and it doesn't send a message if the `intro_message` field is blank

**Comment by Kileak** [comment]
Looks to me as if it would spam intro, if multiple members are invited at once

**Comment by Kileak** [comment]
Finally got time to look at this.

There is already an `intro` command in `BotHandler` (which reads the intro message from `intro_msg`).

I don't want two different sources for an intro message, so this should be changed too (it's ok, if BotHandler would then also read this from the config).

**Comment by crclark96** [comment]
I updated the intro command to read from the `config.json` file as well

File: handlers/bot_handler.py [hunk 2]
```diff
@@ -24,8 +25,10 @@ class IntroCommand(Command):
     def execute(cls, slack_wrapper, args, timestamp, channel_id, user_id, user_is_admin):
         """Execute the Intro command."""
         try:
-            with open("intro_msg") as f:
-                message = f.read()
+            with open("./config.json") as f:
+                message = json.load(f)["intro_message"]
```
Comment (Grazfather) [inline handlers/bot_handler.py:29]: can we use `json.load(f).get("intro_message", "")` so that we don't need to specify the config?

Comment (crclark96) [reply handlers/bot_handler.py]: I changed this - the bot now doesn't respond to the `!intro` command if there isn't a message specified in the config (or if `intro_msg` isn't defined in the config)

File: handlers/bot_handler.py [hunk 3]
```diff
@@ -23,16 +24,10 @@ class IntroCommand(Command):
...
-            with open("intro_msg") as f:
-                message = f.read()
-
-            slack_wrapper.post_message(channel_id, message)
-        except:
-            message = "Sorry, I forgot what I wanted to say (or the admins forgot to give me an intro message :wink:)"
-
-            slack_wrapper.post_message(channel_id, message)
+        with open("./config.json") as f:
+            message = json.load(f).get("intro_message")
```
Comment (Grazfather) [inline handlers/bot_handler.py:28]: Have you tested this without an intro_message set? if it isn't set at all then `message` will be `None`.

We can either force us to set it (don't use `.get`) or allow it to be unset, assume empty, in which case just confirm that you tested it 'cause I am not sue if `post_message` works with `None`

**Comment by crclark96** [comment]
I tested it without a message set, it simply doesn't send any message

**Comment by Grazfather** [comment]
You mean the message field completely missing from the config? if so, great.

You just need to rebase/resolve the conflict.

**Comment by crclark96** [comment]
Yep - with `NoneType`, `parse_message` does nothing.

Rebased.

**Comment by Grazfather** [comment]
Thanks!

**Pull request merged**